### ISSUE TITLE
No lexer found for xonsh in jupyter console

### ISSUE THREAD

**Issue by asmeurer** [xonsh/xonsh#1440]
```
$jupyter console --kernel xonsh
/Users/aaronmeurer/anaconda3/envs/ipython5/lib/python3.5/site-packages/jupyter_console/ptshell.py:94: UserWarning: No lexer found for language 'xonsh'. Treating as plain text.
  warn("No lexer found for language %r. Treating as plain text." % name)
Jupyter console 5.0.0

Xonsh - the Python-ish, BASHwards-looking shell


In [1]:
```

It seems to work in the notebook.

**Comment by scopatz** [comment]
Thanks for reporting @asmeurer.  I see the same issue on my machine.  This seems like a bug with Jupyter console, rather than with xonsh.  Or perhaps it is a bug with ipykernel.  Maybe @Carreau knows more.  In any event, this causes the whole terminal session to hang so I will open an issue on console.

**Comment by asmeurer** [comment]
FYI @gforsyth and I agreed that jupyter console issues should be treated as very low priority. It seems quite buggy.

**Comment by scopatz** [comment]
Sure, but it seems like this could the sort of thing that affects all alternate kernels.

**Comment by scopatz** [comment]
So it is worth reporting upstream.

**Comment by scopatz** [comment]
This has been fixed (theoretically) in ipython/ipykernel#162. I am closing for now, but if the problem persists past ipykernel v4.3.1 and ipython v5.0.0, please reopen.
